core/src/main/java/com/sequenceiq/cloudbreak/service/stack/SharedServiceValidator.java

```diff
@@ -0,0 +1,94 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.model.rds.RdsType;
+import com.sequenceiq.cloudbreak.api.model.stack.StackRequest;
+import com.sequenceiq.cloudbreak.controller.validation.ValidationResult;
+import com.sequenceiq.cloudbreak.controller.validation.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.domain.workspace.Workspace;
+import com.sequenceiq.cloudbreak.service.rdsconfig.RdsConfigService;
+
+@Component
+public class SharedServiceValidator {
+
+    @Inject
+    private RdsConfigService rdsConfigService;
+
+    @Inject
+    private StackViewService stackViewService;
+
+    public ValidationResult checkSharedServiceStackRequirements(StackRequest request, Workspace workspace) {
+        ValidationResultBuilder resultBuilder = ValidationResult.builder();
+        if (request.getClusterToAttach() != null) {
+            checkCloudPlatform(request, resultBuilder);
+            checkSharedServiceRequirements(request, workspace, resultBuilder);
+        }
+        return resultBuilder.build();
+    }
+
+    private void checkCloudPlatform(StackRequest request, ValidationResultBuilder resultBuilder) {
+        Optional<StackView> datalakeStack = stackViewService.findById(request.getClusterToAttach());
+        if (!datalakeStack.isPresent()) {
+            resultBuilder.error("Datalake stack with the requested ID (in clusterToAttach field) was not found.");
```

Comment:
can we use the name here?

Reply:
no, because the stack with the given ID does not exist, therefore it does not have name either.